### ISSUE TITLE
Sendkeys to focussed element should not reset carat position

### ISSUE THREAD

**Issue by stanleyhon** [w3c/webdriver#1430]
There was some discussion on the selenium repository https://github.com/SeleniumHQ/selenium/issues/7330 about the spec specifying

1. Let current text length be the JavaScript string’s length of element’s API value.
2. Set the text insertion caret using set selection range using current text length for both the start and end parameters.

This will stop a scenario where a use attempts to modify the carat position from working correctly such as

- SendKeys "abc"
- SendKeys left arrow 3 times
- Sendkeys "123"

the carat repositioning would be removed.

Based on the selenium thread, it seems like Firefox already does not reset the caret position if the field is in focus.

**Comment by thejohnjansen** [comment]
hey @shs96c  and @AutomatedTester , this is from the discussion we had on hangouts the other day. I could not push my proposed changes to the spec due to permissions, so this issue is meant to drive the spec change necessary for the WebDriver change to merge. 

I do think it's as simple as adding a new step that says something like:

"<li><p>If <var>element</var> already has focus, 
      do not modify the text insertion caret.</p>"

**Comment by AutomatedTester** [comment]
@thejohnjansen sounds good! Please make sure there are wdspec tests to match

**Comment by thejohnjansen** [comment]
I just added one test - I was surprised to see we didn't test this already, nor do we test caret placement in the Perform_Actions tests. 

https://github.com/web-platform-tests/wpt/pull/17664

**Comment by KitoCoding** [comment]
So for clarification, if an element already has focus, don't reset its focus correct? If the index caret is already set to a position, we won't be updating? What If i wanted to update that caret to the end, or a specific location in the text?

**Comment by stanleyhon** [comment]
> So for clarification, if an element already has focus, don't reset its focus correct? If the index caret is already set to a position, we won't be updating?

Yes I believe this to be the intent. See the original selenium bug https://github.com/SeleniumHQ/selenium/issues/7330. Note GeckoDriver reports carat is not reset if the element is already in focus.

> What If i wanted to update that caret to the end, or a specific location in the text?

By entering text normally without sending special codes like "left arrow" you would always insert at the end. E.g.

SendKeys("A")
SendKeys("B")
SendKeys("C")

I would expect to result in "ABC" in the field.

For a specific place, you can either simulate that via sending the right combination of "left arrow" "right arrow" codes or use something else like execute script to use javascript to set the carat exactly where you want it.

**Comment by KitoCoding** [comment]
Thank you for clarifying
